cmd/containerd-shim-runc-v2/manager/manager_linux.go

```diff
@@ -103,6 +104,7 @@ func newCommand(ctx context.Context, id, containerdAddress, containerdTTRPCAddre
 	cmd := exec.Command(self, args...)
 	cmd.Dir = cwd
 	cmd.Env = append(os.Environ(), "GOMAXPROCS=4")
+	cmd.Env = append(cmd.Env, fmt.Sprintf("XDG_RUNTIME_DIR=%s", defaults.DefaultStateDir))
```

Comment:
This is incorrect. `XDG_RUNTIME_DIR` has to follow the XDG spec and must not be overridden with containerd's state dir.

Reply:
how about my new pr? 
```
	if os.Getenv("XDG_RUNTIME_DIR") == "" {
		cmd.Env = append(cmd.Env, fmt.Sprintf("/run/user/%d", os.Getuid()))
	}
```